Mage/src/main/java/mage/abilities/effects/ContinuousEffects.java

```diff
@@ -947,7 +944,19 @@ public synchronized void apply(Game game) {
         removeInactiveEffects(game);
         List<ContinuousEffect> activeLayerEffects = getLayeredEffects(game); // main call
 
-        List<ContinuousEffect> layer = filterLayeredEffects(activeLayerEffects, Layer.CopyEffects_1);
+        List<ContinuousEffect> layer = filterLayeredEffects(activeLayerEffects, Layer.TransformEffects_0);
+        for (ContinuousEffect effect : layer) {
+            Set<Ability> abilities = layeredEffects.getAbility(effect.getId());
+            for (Ability ability : abilities) {
+                effect.apply(Layer.TransformEffects_0, SubLayer.NA, ability, game);
+            }
+        }
+        //Reload layerEffect if transform effects were applied
+        if (!layer.isEmpty()) {
+            activeLayerEffects = getLayeredEffects(game);
+        }
+
+        layer = filterLayeredEffects(activeLayerEffects, Layer.CopyEffects_1);
```

Comment:
Deep game engine changes like that (without tests and detailed description in code and PR) smells very bad.

Why you need all that code instead usage of existing one (there are many other implementations with same "becomes attached to"). Even [[Sanctuary Blade]] already existed and worked in old code.

Reply:
Thanks for the feedback.
This change is because the copy effect needed for [[Dinosaur Headdress]] is currently in the same layer as the transform effect. Since there is only one pass of that layer, the copy effect (`DinosaurHeaddressEffect`) linked with the card on the battlefield does not get added to the layered effects until after the transform effect is processed.

The active layered effects are then re-populated before processing layer 2. This is why non-copy effects work on existing transformed equipment.

[[Sanctuary Blade]] is not the reason for the engine change. However, it currently causes a trigger when it is attached to a creature. This is not correct - it should not allow a response before resolution of the choice.